email-client/src/main/java/uk/gov/hmcts/cmc/email/sendgrid/SendGridClient.java

```diff
@@ -0,0 +1,97 @@
+package uk.gov.hmcts.cmc.email.sendgrid;
+
+import com.sendgrid.Method;
+import com.sendgrid.Request;
+import com.sendgrid.Response;
+import com.sendgrid.SendGrid;
+import com.sendgrid.helpers.mail.Mail;
+import com.sendgrid.helpers.mail.objects.Attachments;
+import com.sendgrid.helpers.mail.objects.Content;
+import com.sendgrid.helpers.mail.objects.Email;
+import org.apache.http.HttpException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.MediaType;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.cmc.email.EmailAttachment;
+import uk.gov.hmcts.cmc.email.EmailData;
+import uk.gov.hmcts.cmc.email.EmailSendFailedException;
+
+import java.io.IOException;
+
+@Component
+public class SendGridClient {
+    private final SendGrid sendGrid;
+
+    @Autowired
+    public SendGridClient(
+        SendGridFactory factory,
+        @Value("${sendgrid.api-key}") String apiKey,
+        @Value("${sendgrid.testing:false}") Boolean testing
+    ) {
+        sendGrid = factory.createSendGrid(apiKey, testing);
+    }
+
+    public void sendEmail(String from, EmailData emailData) throws IOException {
+        verifyData(from, emailData);
+
+        Email sender = new Email(from);
+        String subject = emailData.getSubject();
+        Email recipient = new Email(emailData.getTo());
+        Content content = new Content(MediaType.TEXT_PLAIN_VALUE, emailData.getMessage());
+        Mail mail = new Mail(sender, subject, recipient, content);
+        if (emailData.hasAttachments()) {
+            emailData.getAttachments().stream()
+                .map(SendGridClient::toSendGridAttachments)
+                .forEach(mail::addAttachments);
+        }
+
+        Request request = new Request();
+        request.setMethod(Method.POST);
+        request.setEndpoint("mail/send");
+        request.setBody(mail.build());
+
+        Response response = sendGrid.api(request);
+        if (response.getStatusCode() / 100 != 2) {
```

Comment:
This seems a convoluted way of saying:
```suggestion
        if (response.getStatusCode() != 200) {
```
I assume there is a reason you did it this way?

Reply:
This matches 2xx, not just 200

Reply:
Of course it does.
Personally I would prefer something a bit more explicit, maybe:
```suggestion
        if (!HttpStatus.valueOf(response.getStatusCode()).is2xxSuccessful()) {
```

Reply:
Yep, nice!